src/plugins/intel_gpu/src/kernel_selector/core/cl_kernels/include/algorithm.cl

```diff
@@ -33,3 +33,13 @@
         }                                                               \
         return first;                                                   \
     }
+
+#if OUTPUT_TYPE_SIZE == 2 //f16
+#define ZERO_FP 0.h
+#define HALF_ONE_FP 0.5h
+#define ONE_FP 1.h
```

Comment:
I think these macro should be removed as the names might be misleading - the values refer to to OUTPUT type, but it's not mentioned anywhere in the name. Moreover, IIRC, jitter creates similar macro for all tensors with respect to the data type:  `${TensorName}_VAL_ONE`, `${TensorName}_VAL_ZERO`. For the half you can add special jit variable in the roi align kernel only.